From 0411f6d5f91ef6713b92f067afd56d2750983fe9 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:11:25 -0400
Subject: [powerpc] powernv: Release the refcount for pci_dev

Message-id: <1410545655-205645-457-git-send-email-dzickus@redhat.com>
Patchwork-id: 94466
O-Subject: [RHEL7 PATCH 456/626] powerpc/powernv: Release the refcount for pci_dev
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 4966bfa1b3347ee75e6d93859a2e8ce9a662390c
Author: Wei Yang <weiyang@linux.vnet.ibm.com>
Date:   Wed Apr 23 10:26:33 2014 +0800

    powerpc/powernv: Release the refcount for pci_dev

    On PowerNV platform, we are holding an unnecessary refcount on a pci_dev, which
    leads to the pci_dev is not destroyed when hotplugging a pci device.

    This patch release the unnecessary refcount.

    Signed-off-by: Wei Yang <weiyang@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/pci-ioda.c b/arch/powerpc/platforms/powernv/pci-ioda.c
index 670ce00..110d0d4 100644
--- a/arch/powerpc/platforms/powernv/pci-ioda.c
+++ b/arch/powerpc/platforms/powernv/pci-ioda.c
@@ -334,7 +334,6 @@ static void pnv_ioda_setup_same_PE(struct pci_bus *bus, struct pnv_ioda_pe *pe)
     pci_name(dev));
    continue;
   }
-  pci_dev_get(dev);
   pdn->pcidev = dev;
   pdn->pe_number = pe->pe_number;
   pe->dma_weight += pnv_ioda_dma_weight(dev);
-- 
1.7.1